File: backend/Services/Parsers/ActorleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class ActorleParser : BasicScoreResultParser
    {
        private ILogger<ActorleParser> _logger;

        public ActorleParser(ILogger<ActorleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => false;
        public override bool Default => false;
        public override string GameName => "Actorle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{GameName} #\\d+ (?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => "Play here: ";
        public override string Url => "https://actorle.com";

        protected override string GetCleanResult(string result, Match parserResults)
        {
            return result.Replace(ExtraContent, string.Empty).Replace(Url, string.Empty).Trim();
        }
    }
}

File: lambdas/src/parsers/actorle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class ActorleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Actorle';
  readonly helpText = null;
  protected readonly extraContent = 'Play here: ';
  readonly url = 'https://actorle.com';
  
  // Matches "Actorle #1417 1/8"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `${this.gameName} #\\d+ (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
  
  protected getCleanResult(result: string, match: RegExpMatchArray): string {
    return result
      .replace(this.extraContent ?? '', '')
      .replace(this.url ?? '', '')
      .trim();
  }
}
